src/helpers/relative-activity.tsx

```diff
@@ -0,0 +1,75 @@
+import { add } from "date-fns";
+import { ColorDefinition } from "./colors";
+import { queryDailyData } from "./query-daily-data";
+import getDayKey from "./get-day-key";
+
+export interface RelativeActivityQueryResult {
+    relativePercent: number;
+    value: number;
+    threshold: number;
+}
+
+export interface RelativeActivityDataType {
+    dailyDataType: string;
+    color?: ColorDefinition;
+    overThresholdColor?: ColorDefinition;
+    threshold?: number | "30DayAverage";
+    label?: string;
+    icon?: React.ReactElement;
+    formatter?: (number: number) => string;
+}
+
+export function queryRelativeActivity(startDate: Date, endDate: Date, dataTypes: RelativeActivityDataType[], preview: boolean): Promise<{ [key: string]: { [key: string]: RelativeActivityQueryResult } }> {
+    let relativeActivityResults: { [key: string]: { [key: string]: RelativeActivityQueryResult } } = {};
+    let promises = dataTypes.map(dataType =>
+        queryDailyData(
+            dataType.dailyDataType,
+            add(startDate, { days: dataType.threshold == undefined ? -31 : -1 }),
+            add(endDate, { days: 1 }),
+            preview));
+    return Promise.all(promises).then((results) => {
+        dataTypes.forEach((dataType, index) => {
+            let dataTypeData = results[index];
+
+            relativeActivityResults[dataType.dailyDataType] = {};
+
+            let currentDate = startDate;
```

Comment:
I don't recall exactly where I saw it, but I just wanted to point out that while functionally the same, I do like how you converted the start date to `currentDate` here.  I think there are a couple other loops elsewhere that are just updating the `startDate` var, which doesn't break the functionality, but just feels a tad off because it is an incoming parameter.  🙂